### ISSUE TITLE
Resize of window triggers infinite loop

### ISSUE THREAD

**Issue by koppa** [arybczak/ncmpcpp#260]
Versions: ncmpcpp stable and master branch affected

When resizing the terminal to a height of  5 lines or less. The program locks up and uses 100% cpu.

**Comment by koppa** [comment]
I found the problem:

In case of small window height an integer underflow occurs and the validation of the window size fails.
This is my quick and dirty fix:

```
diff --git a/src/actions.cpp b/src/actions.cpp
index e94fe29..34bc5ac 100644
--- a/src/actions.cpp
+++ b/src/actions.cpp
@@ -218,7 +218,9 @@ void resizeScreen(bool reload_main_window)
                getch();
        }
 
-       MainHeight = LINES-(Config.design == Design::Alternative ? 7 : 4);
+       MainHeight = std::max(
+                       (int) LINES - (Config.design == Design::Alternative ? 7 : 4),
+                       0);
 
        validateScreenSize();
 
@@ -257,7 +259,9 @@ void setWindowsDimensions()
        using Global::MainHeight;
        
        MainStartY = Config.design == Design::Alternative ? 5 : 2;
-       MainHeight = LINES-(Config.design == Design::Alternative ? 7 : 4);
+       MainHeight = std::max(
+                       (int) LINES - (Config.design == Design::Alternative ? 7 : 4),
+                        0);
        
        if (!Config.header_visibility)
        {
```

**Comment by arybczak** [comment]
Fixed via db619c6159061d5abd50c18fc2873d66e2452c87, thanks for pinpointing the problem.
